cli/cmd/plugin/unmanaged-cluster/cluster/minikube.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/config"
+)
+
+var minikubeBinName = "minikube"
+
+type MinikubeClusterManager struct{}
+
+// Create
+func (mkcm MinikubeClusterManager) Create(c *config.UnmanagedClusterConfig) (*KubernetesCluster, error) {
+	if !mkcm.isMinikubeRunning() {
+		// TODO: Is minikube it the current context? If not, attempt to switch to it
+
+		args := []string{
+			"start",
+		}
+
+		cmd := exec.Command(minikubeBinName, args...)
+
+		err := cmd.Run()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// readkubeconfig in bytes from configured kubeconfig.
+	// During `Prepare` this should be set to the users home kubeconfig since that's what minikube automatically uses
+	kcBytes, err := os.ReadFile(c.KubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	kc := &KubernetesCluster{
+		Name:       c.ClusterName,
+		Kubeconfig: kcBytes,
+	}
+
+	return kc, nil
+}
+
+// isMinikubeRunning returns true if `minikube status` is successful
+// Running `minikube status` returns an error if the node is stopped or no profile is found
+// In those cases, we can attempt to recreate
+func (mkcm MinikubeClusterManager) isMinikubeRunning() bool {
```

Comment:
With the change in https://github.com/vmware-tanzu/community-edition/pull/3674/files#r833836516, I believe this can largely go away.

Meaning, if we run a create command and the name matches a cluster profile that pre-exists, we should fail.